src/Text/Pandoc/Parsing.hs

```diff
@@ -205,18 +208,30 @@ type Parser t s = Parsec t s
 
 type ParserT = ParsecT
 
-newtype F a = F { unF :: Reader ParserState a } deriving (Monad, Applicative, Functor)
+-- | Reader monad wrapping the parser state. This is used to possibly delay
+-- evaluation until all relevant information has been parsed and made available
+-- in the parser state.
+newtype Delayable s a = Delayable { runDelayed :: Reader s a }
```

Comment:
Not sure about the nomenclature here.  `Delayable` sounds more like the name of a type class than a type.
My name `F` was meant to stand for `Future`, so perhaps `Future` or `Delayed` would make more sense?
Note that names like `askF` make a lot more sense if it's `F` or `Future`.